### PR TITLE
update refactorization task

### PR DESCRIPTION
**Pull request by sperBruno**

### CODE REVIEW

File: FirstPracticeSelenium/.classpath [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (carledriss) [inline FirstPracticeSelenium/.classpath]: This file is not necessary
Please add it in the .gitignore file and remove it.

File: FirstPracticeSelenium/.project [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (carledriss) [inline FirstPracticeSelenium/.project]: This file is not necessary
Please add it in the .gitignore file and remove it.

File: FirstPracticeSelenium/src/Main.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import movies.Customer;
+import movies.Movie;
+import movies.Rental;
+
+public class Main {
+
+	public static void main(String[] args) {
```
Comment (carledriss) [inline FirstPracticeSelenium/src/Main.java]: Remove this unnecessary class

File: FirstPracticeSelenium/src/main/java/com/fundacionjala/movies/Main.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.fundacionjala.movies;
+
+public class Main {
```
Comment (carledriss) [inline FirstPracticeSelenium/src/main/java/com/fundacionjala/movies/Main.java]: Remove this unnecessary class

File: FirstPracticeSelenium/src/main/java/com/fundacionjala/movies/Movie.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
+package com.fundacionjala.movies;
+
+public  class Movie {
+    public static final int CHILDREN = 2;
+    public static final int REGULAR = 0;
+    public static final int NEW_RELEASE = 1;
```
Comment (carledriss) [inline FirstPracticeSelenium/src/main/java/com/fundacionjala/movies/Movie.java]: Is it necessary to use those constants?

File: FirstPracticeSelenium/src/main/java/com/fundacionjala/movies/Movie.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    private String title;
+    private int priceCode;
```
Comment (carledriss) [inline FirstPracticeSelenium/src/main/java/com/fundacionjala/movies/Movie.java]: Is it necessary to use the priceCode

File: FirstPracticeSelenium/src/movies/Customer.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
+package movies;
+
+import java.util.Enumeration;
+import java.util.Vector;
+
+public class Customer {
```
Comment (carledriss) [inline FirstPracticeSelenium/src/movies/Customer.java]: This class is the old one

**Comment by carledriss** [comment]
@sperBruno 
Please update your PR
you still have old classes

**Comment by sperBruno** [comment]
@carledriss please review the last commit of refactoring movies project

**Comment by carledriss** [comment]
@sperBruno 
hereditage ????????
Review the PR name

**Comment by carledriss** [comment]
@sperBruno 
Please update your PR with the last changes that we did

Don't forget to add javadocs

File: Refactoring/main/java/com/fundacionjala/movies/Constant.java [hunk 8]
```diff
@@ -0,0 +1,7 @@
+package com.fundacionjala.movies;
+
+
+public class Constant {
```
Comment (carledriss) [inline Refactoring/main/java/com/fundacionjala/movies/Constant.java]: Please add a private constructor for classes that contains constants and static methods

File: Refactoring/main/java/com/fundacionjala/movies/Customer.java [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+    private final String name;
+
+    private List<Rental> rentals = new ArrayList<>();
+
+    Customer(String name) {
+        this.name = name;
+    }
+
+    void addRental(Rental rental) {
```
Comment (carledriss) [inline Refactoring/main/java/com/fundacionjala/movies/Customer.java]: This method should be public

File: Refactoring/main/java/com/fundacionjala/movies/Customer.java [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+        rentals.add(rental);
+    }
+
+    /**
+     * This method calculates the total charge that a user has to pay for the rented movies.
+     *
+     * @return the total amount of money that a user has to pay.
+     */
+    private double calculateTotalCharge() {
```
Comment (carledriss) [inline Refactoring/main/java/com/fundacionjala/movies/Customer.java]: This method should be public

File: Refactoring/main/java/com/fundacionjala/movies/Customer.java [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+        }
+        return totalCharge;
+    }
+
+    /**
+     * This method calculates the frequent points that a user gets for the rented movies.
+     *
+     * @return the frequent renter points that a user get.
+     */
+    private int calculateTotalFrequentRenterPoints() {
```
Comment (carledriss) [inline Refactoring/main/java/com/fundacionjala/movies/Customer.java]: This method should be public

File: Refactoring/main/java/com/fundacionjala/movies/Customer.java [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+        }
+        return frequentRenterPoints;
+    }
+
+    /**
+     * This method is used to print the ticket of the price of the rented movies.
+     *
+     * @return the resume of the price of the rented movies.
+     */
+    String statement() {
```
Comment (carledriss) [inline Refactoring/main/java/com/fundacionjala/movies/Customer.java]: This method should be public

File: Refactoring/main/java/com/fundacionjala/movies/Customer.java [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+        result.append("Amount owed is ");
+        result.append(calculateTotalCharge());
+        result.append(BREAK_LINE);
+        result.append("You earned ");
+        result.append(calculateTotalFrequentRenterPoints());
+        result.append(" frequent renter points");
+        return result.toString();
+    }
+
+    String getName() {
```
Comment (carledriss) [inline Refactoring/main/java/com/fundacionjala/movies/Customer.java]: This method should be public

File: Refactoring/main/java/com/fundacionjala/movies/Customer.java [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+        return name;
+    }
+
+    List<Rental> getRentals() {
```
Comment (carledriss) [inline Refactoring/main/java/com/fundacionjala/movies/Customer.java]: This method should be public

File: Refactoring/main/java/com/fundacionjala/movies/Movie.java [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+
+    private final String title;
+
+    private static final double DISCOUNT_CHARGE = 1.5;
+
+    private final static int DAYS_RENTED = 1;
+
+    private final static int NEW_RELEASE = 1;
+
+    private static int FREQUENT_RENTER_POINTS = 1;
```
Comment (carledriss) [inline Refactoring/main/java/com/fundacionjala/movies/Movie.java]: this variable should be final

File: Refactoring/main/java/com/fundacionjala/movies/Movie.java [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+
+    private final int priceCode;
+
+    private final String title;
+
+    private static final double DISCOUNT_CHARGE = 1.5;
+
+    private final static int DAYS_RENTED = 1;
+
+    private final static int NEW_RELEASE = 1;
```
Comment (carledriss) [inline Refactoring/main/java/com/fundacionjala/movies/Movie.java]: This variable is not necessary.
Delete it.

File: Refactoring/main/java/com/fundacionjala/movies/Movie.java [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+        return (daysRented > DAYS_LIMIT) ? (daysRented - DAYS_LIMIT) * DISCOUNT_CHARGE : DISCOUNT_CHARGE;
+    }
+
+    /**
+     * This method calculate the point that a users get for their rents.
+     *
+     * @param daysRented this parameter is used to calculated how many points a user will receive
+     * @return the points that user get for the rent.
+     */
+    int calculateFrequentRenterPoints(int daysRented) {
```
Comment (carledriss) [inline Refactoring/main/java/com/fundacionjala/movies/Movie.java]: this should be public

File: Refactoring/main/java/com/fundacionjala/movies/Movie.java [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+        return ((priceCode == NEW_RELEASE)
+                && daysRented > DAYS_RENTED) ? FREQUENT_RENTER_POINTS++ : FREQUENT_RENTER_POINTS;
+    }
+
+    int getPriceCode() {
+        return priceCode;
+    }
+
+    String getTitle() {
```
Comment (carledriss) [inline Refactoring/main/java/com/fundacionjala/movies/Movie.java]: this should be public

File: Refactoring/main/java/com/fundacionjala/movies/Movie.java [hunk 19]
```diff
@@ -0,0 +1,57 @@
+package com.fundacionjala.movies;
+
+/**
+ * This an abstract class that is used by Regular, Children and NewRelease classes.
+ */
+
+abstract class Movie {
+
+    private final int priceCode;
```
Comment (carledriss) [inline Refactoring/main/java/com/fundacionjala/movies/Movie.java]: Is it necessary this attribute?
keep in mind you are using now inheritance.
Delete it.

File: Refactoring/main/java/com/fundacionjala/movies/Movie.java [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+     *
+     * @param daysRented this parameter is used to calculated how many points a user will receive
+     * @return the points that user get for the rent.
+     */
+    int calculateFrequentRenterPoints(int daysRented) {
+        return ((priceCode == NEW_RELEASE)
+                && daysRented > DAYS_RENTED) ? FREQUENT_RENTER_POINTS++ : FREQUENT_RENTER_POINTS;
+    }
+
+    int getPriceCode() {
```
Comment (carledriss) [inline Refactoring/main/java/com/fundacionjala/movies/Movie.java]: Is this method necessary?
keep in mind you are using now inheritance.
Delete it.

File: Refactoring/test/java/com/fundacionjala/movies/TestRegular.java [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+        assertEquals(expectedCharge, rental.calculateCharge(), Constant.DELTA);
+    }
+
+    /**
+     * this test verifies the correct charge when the days rented is bigger than 3 days.
+     */
+    @Test
+    public void validateChargeWhenDaysRenterBiggerThan3() {
+        Rental rental = new Rental(new Regular("Batman", 5), 5);
+        double expectedCharge = 4.5;
```
Comment (carledriss) [inline Refactoring/test/java/com/fundacionjala/movies/TestRegular.java]: This variable should be final

File: Refactoring/test/java/com/fundacionjala/movies/TestRegular.java [hunk 22]
```diff
@@ -0,0 +1,51 @@
...
+ */
+public class TestRegular {
+
+    /**
+     * this test verifies the correct charge when days rented is less than 3 days.
+     */
+    @Test
+    public void validateCharge() {
+        Rental rental = new Rental(new Regular("Batman", 5), 3);
+        double expectedCharge = 1.5;
```
Comment (carledriss) [inline Refactoring/test/java/com/fundacionjala/movies/TestRegular.java]: This variable should be final

File: Refactoring/test/java/com/fundacionjala/movies/TestRegular.java [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+        assertEquals(expectedCharge, rental.calculateCharge(), Constant.DELTA);
+    }
+
+    /**
+     * this test verifies the correct frequent renter days.
+     */
+    @Test
+    public void calculateFrequentRenterPoints() {
+        Rental rental = new Rental(new Regular("Batman", 2), 5);
+        int expectedCharge = 3;
```
Comment (carledriss) [inline Refactoring/test/java/com/fundacionjala/movies/TestRegular.java]: This variable should be final

**Comment by sperBruno** [comment]
@carledriss please review the last commit

File: Refactoring/test/java/com/fundacionjala/movies/ChildrenTest.java [hunk 24]
```diff
@@ -0,0 +1,51 @@
...
+ * This class is used to verify the  corectness of the Children class.
+ */
+public class ChildrenTest {
+
+    /**
+     * this test verifies the correct charge when days rented is less than 3 days.
+     */
+    @Test
+    public void validateCharge() {
+        Rental rental = new Rental(new Children("Madagascar"), 3);
```
Comment (carledriss) [inline Refactoring/test/java/com/fundacionjala/movies/ChildrenTest.java]: This line is repeated in all tests.
Please add a @Before method

File: Refactoring/test/java/com/fundacionjala/movies/CostumerTest.java [hunk 25]
```diff
@@ -0,0 +1,64 @@
...
+import org.junit.Test;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * This class is used to validate the charge and the frequent renter points resume
+ * of customer's rent
+ */
+public class CostumerTest {
```
Comment (carledriss) [inline Refactoring/test/java/com/fundacionjala/movies/CostumerTest.java]: Typo
Customer

File: Refactoring/test/java/com/fundacionjala/movies/NewReleaseTest.java [hunk 26]
```diff
@@ -0,0 +1,39 @@
...
+ * This class is used to validate the charge and the frequent renter points of New Release movie.
+ */
+public class NewReleaseTest {
+
+    /**
+     * This method validate the amount of charge for New Release movie.
+     */
+    @Test
+    public void validateCharge() {
+        Rental rental = new Rental(new NewRelease("Madagascar"), 3);
```
Comment (carledriss) [inline Refactoring/test/java/com/fundacionjala/movies/NewReleaseTest.java]: This line is repeated in all tests
Please add a @Before method

File: Refactoring/test/java/com/fundacionjala/movies/RegularTest.java [hunk 27]
```diff
@@ -0,0 +1,49 @@
...
+ * This class is used to validate the charge and the frequent renter points of Regular movie.
+ */
+public class RegularTest {
+
+    /**
+     * this test verifies the correct charge when days rented is less than 3 days.
+     */
+    @Test
+    public void validateCharge() {
+        Rental rental = new Rental(new Regular("Batman"), 3);
```
Comment (carledriss) [inline Refactoring/test/java/com/fundacionjala/movies/RegularTest.java]: This line is repeated in all tests
Please add a @Before method

**Comment by sperBruno** [comment]
@carledriss  please review the last commit.
thanks for your time.

File: Refactoring/main/java/com/fundacionjala/movies/Constant.java [hunk 28]
```diff
@@ -9,6 +9,10 @@
 
     public static final int EXPECTED_POINTS = 2;
 
+    public static Rental RENTAL;
```
Comment (carledriss) [inline Refactoring/main/java/com/fundacionjala/movies/Constant.java]: This attribute is not a constant.
Remove it!

File: Refactoring/main/java/com/fundacionjala/movies/Constant.java [hunk 29]
```diff
@@ -9,6 +9,10 @@
...
+
+    public static double EXPECTED_CHARGE;
```
Comment (carledriss) [inline Refactoring/main/java/com/fundacionjala/movies/Constant.java]: This attribute is not a constant.
Remove it!

**Comment by sperBruno** [comment]
@carledriss sorry to bother you, please review the last commit. Thanks for your time

**Comment by carledriss** [comment]
Looks good to me!

**Pull request merged**